ats/machines.py

```diff
@@ -317,7 +317,8 @@ def testEnded(self, test, status):
            #note test.status is not necessarily status after this!
            #see test.expectedResult
 
-        self.noteEnd(test)  #to be defined in children
+        if not configuration.options.removeEndNote:
+            self.noteEnd(test)  #to be defined in children
```

Comment:
I don't think this is correct because some machines (slurm, flux, and I think lsf) use noteEnd to calculate the number of remaining resources.  The implementations of noteEnd need to be refactored to remove the print portion (perhaps as logEnd).

Reply:
Check, thanks for that @liu15

Reply:
you are correct Ben.  That needs fixed.